spec/models/schema_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe Schema, type: :model do
+  context 'when created' do
+    subject { described_class.new }
+
+    it 'must have an endpoint and a structure' do
+      expect { subject.save! }.to raise_error(ActiveRecord::RecordInvalid)
+      subject.structure = '{ "field": "value" }'
+      expect { subject.save! }.to raise_error(ActiveRecord::StatementInvalid)
+    end
+
+    it 'must have a non-empty structure' do
+      subject.structure = '{ }'
+      expect { subject.save! }.to raise_error(ActiveRecord::RecordInvalid)
+    end
```

Comment:
Yeah, I'm more trying to make sure that the invariants on my models hold.  I'm less worried about the relational integrity tests (e.g. has an endpoint) and more about the expectations on the JSON data and the effective `tsrange`.